cpp/client/ct.cc

```diff
@@ -753,7 +755,7 @@ static void DiagnoseCertChain() {
       continue;
     }
     LOG(INFO) << "SCT number " << i + 1 << ":\n" << sct.DebugString();
-    if (verifier != NULL) {
+    if (verifier.get()) {
```

Comment:
On Fri, 21 Aug 2015 at 22:36 Pierre Phaneuf <notifications@github.com>
wrote:

> In cpp/client/ct.cc
> <https://github.com/google/certificate-transparency/pull/874#discussion_r37680673>
> :
>
> > @@ -753,7 +755,7 @@ static void DiagnoseCertChain() {
> >        continue;
> >      }
> >      LOG(INFO) << "SCT number " << i + 1 << ":\n" << sct.DebugString();
> > -    if (verifier != NULL) {
> > +    if (verifier.get()) {
>
> I don't know why, I had a vague feeling you might have had some Control
> Data in your past life... [image: :smiley:]
>
> http://c-faq.com/null/machexamp.html
>

Sorry to disappoint, but the CDC 6400 had 18 bit pointers. And I'm pretty
sure null would've been zero.


> If you're looking to go against this if (p) style, you've got a solid
> uphill battle ahead of you... For example, because bool is an integer
> (and so, will auto-convert to int and friends), objects that want to be
> "testable" aren't wise to implement an operator bool, but instead have a
> conversion operator to some non-void pointer (usually a private inner class
> that's never defined, or a method pointer, since those won't bloody convert
> even to things they ought to!).
>
Hmm. That seems like a different thing, though.


> I guess those objects would then be comparable with nullptr, say, but
> you're going to have a big pile of existing code that does this (and is
> perfectly correct to do it).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/874/files#r37680673>
> .
>